src/tasks/tasks.py

```diff
@@ -1579,3 +1586,196 @@ def load_data(self, path, max_seq_len):
         self.val_data_text = val_data
         self.test_data_text = te_data
         log.info('\tFinished loading CCGTagging data.')
+
+
+class SpanTask(Task):
+    ''' Generic class for span tasks.
+    Acts as a classifier, but with multiple targets for each input text.
+    Targets are of the form (span1, span2,..., span_n, label), where the spans are
+    half-open token intervals [i, j).
+    '''
+    @property
+    def _tokenizer_suffix(self):
```

Comment:
So for span tokenization, right now it is a 2-step process. You have to retokenize a file into json format (which I think is honestly better than the .tsv format we have for non-span tasks because it is more structured). There is additional trickiness with spans because you have to re-align span indicies after tokenization so I do that as a preprocessing step to avoid computational burden during run-time.